### PR TITLE
steamPackages.steamcmd: add mainProgram, fix fetch url to use an archive

### PR DESCRIPTION
**Pull request by diniamo**
## Description of changes

Adds the missing mainProgram meta attribute to steamPackages.steamcmd.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by Qyriad — APPROVED** [review]

File: pkgs/games/steam/steamcmd.nix [hunk 1]
```diff
@@ -37,8 +37,9 @@ stdenv.mkDerivation {
   '';
 
   meta = with lib; {
-    description = "Steam command-line tools";
     homepage = "https://developer.valvesoftware.com/wiki/SteamCMD";
+    description = "Steam command-line tools";
```
Comment (Aleksanaa) [inline pkgs/games/steam/steamcmd.nix:41]: Why `description` has to be under `homepage`? (I mean why are you making this change)

Comment (diniamo) [reply pkgs/games/steam/steamcmd.nix:41]: It doesn't matter, but that's the order they're in in most derivations

Comment (SuperSandro2000) [reply pkgs/games/steam/steamcmd.nix:41]: Yeah, just leave it. I would have ignored it but I found another issue which we should fix.

**Review by SuperSandro2000 — COMMENTED** [review]
src should be fetched via web.archive.org otherwise we might get a wrong hash when they ever update it.

Also we can switch to hash while we are at it.

https://web.archive.org/web/20240521141411/https://steamcdn-a.akamaihd.net/client/installer/steamcmd_linux.tar.gz

**Comment by diniamo** [comment]
@SuperSandro2000 should I leave the version as is, or should I make it the number in the url, and use that to fetch?

File: pkgs/games/steam/steamcmd.nix [hunk 2]
```diff
@@ -7,8 +7,8 @@ stdenv.mkDerivation {
   version = "20180104"; # According to steamcmd_linux.tar.gz mtime
 
   src = fetchurl {
-    url = "https://steamcdn-a.akamaihd.net/client/installer/steamcmd_linux.tar.gz";
-    sha256 = "0z0y0zqvhydmfc9y9vg5am0vz7m3gbj4l2dwlrfz936hpx301gyf";
+    url = "https://web.archive.org/web/20240521141411/https://steamcdn-a.akamaihd.net/client/installer/steamcmd_linux.tar.gz";
+    sha256 = "sha256-zr8ARr/QjPRdprwJSuR6o56/QVXl7eQTc7V5uPEHHnw=";
```
Comment (SuperSandro2000) [inline pkgs/games/steam/steamcmd.nix:11]:
```suggestion
    hash = "sha256-zr8ARr/QjPRdprwJSuR6o56/QVXl7eQTc7V5uPEHHnw=";
```

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
Just built the package out of curiosity and the old hash was already wrong... #317317

**Comment by diniamo** [comment]
Huhh, I updated the hash though.

**Comment by SuperSandro2000** [comment]
It's the same hash, just in SRI

**Comment by diniamo** [comment]
What's SRI?

**Comment by SuperSandro2000** [comment]
see https://wiki.nixos.org/wiki/Nix_Hash

**Pull request merged**